internal/nvmeof/nvmeof_initiator.go

```diff
@@ -0,0 +1,194 @@
+
+	//
+	GetNamespaceDeviceByUUID(ctx context.Context, uuid string) (string, error)
+}
+
+// ConnectRequest represents a subsystem connection request.
+type ConnectRequest struct {
+	SubsystemNQN string
+	// TODO - change to struct with ip + port that exist in the current unmerged PR.
+	Addresses []string // ["10.242.64.32", "10.242.64.33"]
+	Ports     []string // ["4420", "4420"]
+	Transport string   // "tcp"
+	HostNQN   string   // Optional - empty means use system default
+}
+
+// nvmeInitiator implements NVMeInitiator interface.
+type nvmeInitiator struct{}
+
+// NewNVMeInitiator creates a new NVMe-oF initiator.
+func NewNVMeInitiator() NVMeInitiator {
+	return &nvmeInitiator{}
+}
+
+// LoadKernelModules ensures required kernel modules are loaded.
+func (ni *nvmeInitiator) LoadKernelModules(ctx context.Context) error {
+	log.DebugLog(ctx, "Loading NVMe-oF kernel modules")
+	var stderr string
+	modules := []string{
+		"nvme_tcp",
+		"nvme_fabrics",
+	}
+
+	for _, module := range modules {
+		// check if the module is loaded or compiled in
+		_, err := os.Stat("/sys/module/" + module)
+		if os.IsNotExist(err) {
+			// try to load the module
+			_, stderr, err = util.ExecCommand(context.TODO(), "modprobe", module)
+			if err != nil {
+				log.DebugLog(ctx, "%s modprobe failed (%v): %q", module, err, stderr)
+			}
+		}
+	}
+
+	log.DebugLog(ctx, "All NVMe-oF kernel modules loaded successfully")
+
+	return nil
+}
+
+// ConnectSubsystem connects to an NVMe-oF subsystem.
+func (ni *nvmeInitiator) ConnectSubsystem(ctx context.Context, req *ConnectRequest) (bool, error) {
+	log.DebugLog(ctx, "Connecting to NVMe-oF subsystem %s at %v:%v",
+		req.SubsystemNQN, req.Addresses, req.Ports)
+
+	// Try connecting to each address until one succeeds
+	var lastErr error
+	for i, address := range req.Addresses {
+		port := "4420" // TODO: add Default port?
+		if i < len(req.Ports) {
+			port = req.Ports[i]
+		}
+
+		// Build nvme connect command for this address
+		args := []string{
+			"connect",
+			"-t", req.Transport,
+			"-n", req.SubsystemNQN,
+			"-a", address,
+			"-s", port,
+			"-l", "1800", // TODO - known value for connection timeout.move to be const.
+		}
+
+		// Add HostNQN only if specified
+		if req.HostNQN != "" {
+			args = append(args, "--hostnqn", req.HostNQN)
+		}
+
+		stdout, stderr, err := util.ExecCommandWithTimeout(ctx, connectTimeout, "nvme", args...)
+		// Execute connection
+		if err != nil {
+			log.WarningLog(ctx, "Failed to connect to %s:%s - stdout: %s, stderr: %s", address, port, stdout, stderr)
+			lastErr = err // TODO: fail or continue to next?
+
+			continue
+		}
+
+		log.DebugLog(ctx, "Successfully connected to subsystem %s via %s:%s",
+			req.SubsystemNQN, address, port)
+
+		return true, nil
+	}
+
+	// TODO: what to do if the list is empty or failed to connect to all ip? error or warning?
+	return false, fmt.Errorf("failed to connect to any gateway: %w", lastErr)
```

Comment:
Failing to connect to all gateways should return an error. Even if one gateway is connected, it would be successful.